src/MonoTorrent.Client/MonoTorrent.Client/Managers/ListenManager.cs

```diff
@@ -121,6 +121,9 @@ async ReusableTask<bool> HandleHandshake (PeerInfo peerInfo, IPeerConnection con
             if (message.ProtocolString != Constants.ProtocolStringV100)
                 return false;
 
+            if (Engine.PeerId.Equals (message.PeerId))
+                return false;
```

Comment:
This moves the peer id check earlier, prior to raising the 'PeerConnected' event and adding the peer as an active connection. It's a little cleaner this way :)